File: crop_logo.py
from PIL import Image
import os

img_path = r"c:\Users\nkash\Downloads\ChatGPT Image May 30, 2026, 02_47_22 AM.png"
img = Image.open(img_path)

# Let's crop the top section (dark background logo with text)
# The image is 1024x1024 typically from ChatGPT (or similar aspect ratio)
width, height = img.size

# Assuming it's a grid, the top dark section is roughly the top 40%
dark_logo = img.crop((0, 0, width, int(height * 0.40)))
dark_logo.save(r"c:\kaarya-os\public\logo-dark.png")

# The light logo is below it, roughly from 40% to 60%, but let's check
light_logo = img.crop((0, int(height * 0.45), int(width * 0.6), int(height * 0.65)))
light_logo.save(r"c:\kaarya-os\public\logo-light.png")

print(f"Original size: {width}x{height}")
print("Saved cropped logos.")

File: crop_logo.js
const sharp = require('sharp');
const path = require('path');

const imgPath = "c:\\Users\\nkash\\Downloads\\ChatGPT Image May 30, 2026, 02_47_22 AM.png";
const darkOutPath = "c:\\kaarya-os\\public\\logo-dark.png";
const lightOutPath = "c:\\kaarya-os\\public\\logo-light.png";

async function crop() {
    try {
        const metadata = await sharp(imgPath).metadata();
        const width = metadata.width;
        const height = metadata.height;

        // Crop top 35% for dark logo (assuming it's at the top)
        await sharp(imgPath)
            .extract({ left: 0, top: 0, width: width, height: Math.floor(height * 0.35) })
            .toFile(darkOutPath);
        
        // Crop middle 35-70% for light logo
        await sharp(imgPath)
            .extract({ left: 0, top: Math.floor(height * 0.35), width: width, height: Math.floor(height * 0.35) })
            .toFile(lightOutPath);
            
        console.log("Successfully cropped logos!");
    } catch (error) {
        console.error("Error cropping image:", error);
    }
}

crop();
